wpilibj/src/main/java/edu/wpi/first/wpilibj/experimental/command/SequentialCommandGroup.java

```diff
@@ -0,0 +1,92 @@
+package edu.wpi.first.wpilibj.experimental.command;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+
+import edu.wpi.first.wpilibj.command.IllegalUseOfCommandException;
+
+public class SequentialCommandGroup extends CommandGroupBase implements Command {
+
+  private final List<Command> m_commands = new ArrayList<>();
+  private Queue<Command> m_commandQueue;
+
+  /**
+   * Creates a new SequentialCommandGroup.  The given commands will be run sequentially, with
+   * the commandgroup finishing when the last command finishes.
+   *
+   * @param commands the commands to include in this group.
+   */
+  public SequentialCommandGroup(Command... commands) {
+    if (!Collections.disjoint(Set.of(commands), getGroupedCommands())) {
+      throw new IllegalUseOfCommandException(
+          "Commands cannot be added to more than one CommandGroup");
+    }
+
+    registerGroupedCommands(commands);
+
+    m_commands.addAll(List.of(commands));
+  }
+
+  @Override
+  public void initialize() {
+    m_commandQueue = new ArrayDeque<>(m_commands);
```

Comment:
Since we just have a list, we should just use an int of current index to avoid the extra allocation, and not have to worry about the deque causing reallocations.

Reply:
Done.